pkgs/top-level/perl-packages.nix

```diff
@@ -15233,6 +15233,10 @@ let
       url = "mirror://cpan/authors/id/L/LE/LEONT/Module-Build-0.4231.tar.gz";
       hash = "sha256-fg9MaSwXQMGshOoU1+o9i8eYsvsmwJh3Ip4E9DCytxc=";
     };
+    postPatch = lib.optionalString (stdenv.hostPlatform != stdenv.buildPlatform) ''
+      # use fullperl since default perl do not support dynamic linking
+      export PATH="$fullperl/bin:$PATH"
```

Comment:
use `addToSearchPath PATH "$fullperl/bin"`

Reply:
Your proposal doesn't work. The function addToSearchPath adds dir at the end off and not at the beginning.